libp2p/NodeTable.cpp

```diff
@@ -190,10 +215,25 @@ shared_ptr<NodeEntry> NodeTable::nodeEntry(NodeID const& _id)
     return it != m_allNodes.end() ? it->second : shared_ptr<NodeEntry>();
 }
 
-void NodeTable::doDiscover(NodeID _node, unsigned _round, shared_ptr<set<shared_ptr<NodeEntry>>> _tried)
+void NodeTable::doDiscover(boost::system::error_code _ec, NodeID _node, unsigned _round,
+    shared_ptr<set<shared_ptr<NodeEntry>>> _tried, shared_ptr<ba::deadline_timer> _timer)
 {
     // NOTE: ONLY called by doDiscovery!
 
+    // We can't use m_logger yet since the node table might have already been destroyed
+    if (_ec == boost::asio::error::operation_aborted ||
+        _timer->expires_at() == boost::posix_time::min_date_time)
```

Comment:
Could you explain when this second condition can happen?

Reply:
We set the min time instead of cancelling to get around the race condition, look in the node table destructor.